### PR TITLE
use different operator for units on xray lines

### LINKED ISSUE #170: Entire package on `main` branch cannot be imported because of unit bug in `legacy.thermal`
### Describe the bug

The units are messed up somewhere in `sunkit_spex.legacy.thermal`, which breaks the import of the whole package

### To Reproduce

First set up a new venv here I'm using `uv`, Python 3.13.0
```bash
cd $(mktemp -d)
uv venv
source .venv/bin/activate

uv pip install git+https://github.com/sunpy/sunkit-spex.git
```

Now execute the import command
```bash
# pipe it to a file if you want
python -c 'import sunkit_spex' # > log.txt
```

Example output
```
Traceback (most recent call last):                                                                                                 
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/astropy/units/quantity.py", line 974, in to_value
    scale = self.unit._to(unit)
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/astropy/units/core.py", line 1168, in _to
    raise UnitConversionError(f"'{self!r}' is not a scaled version of '{other!r}'")
astropy.units.errors.UnitConversionError: 'Unit("keV2")' is not a scaled version of 'Unit("keV")'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
    import sunkit_spex
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/sunkit_spex/__init__.py", line 3, in <module>
    from . import extern
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/sunkit_spex/extern/__init__.py", line 9, in <module>
    from . import rhessi, stix
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/sunkit_spex/extern/rhessi.py", line 12, in <module>
    from sunkit_spex.legacy.fitting import instruments
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/sunkit_spex/legacy/fitting/__init__.py", line 1, in <module>
    from . import (
    ...<9 lines>...
    )
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/sunkit_spex/legacy/fitting/fitter.py", line 43, in <module>
    from sunkit_spex.legacy.fitting.photon_models_for_fitting import defined_photon_models  # noqa
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/sunkit_spex/legacy/fitting/photon_models_for_fitting.py", line 13, in <module>
    from sunkit_spex.legacy.thermal import thermal_emission
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/sunkit_spex/legacy/thermal.py", line 191, in <module>
    LINE_GRID = setup_line_parameters()
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/sunkit_spex/legacy/thermal.py", line 157, in setup_line_parameters
    line_grid["line peaks keV"] = (line_info.peak_energy.data * line_info.attrs["units"]["peak_energy"]).to_value(
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        u.keV, equivalencies=u.spectral()
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/astropy/units/quantity.py", line 977, in to_value
    value = self._to_value(unit, equivalencies)
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/astropy/units/quantity.py", line 883, in _to_value
    return self.unit.to(
           ~~~~~~~~~~~~^
        unit, self.view(np.ndarray), equivalencies=equivalencies
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/astropy/units/core.py", line 1204, in to
    return self.get_converter(Unit(other), equivalencies)(value)
           ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/astropy/units/core.py", line 1133, in get_converter
    raise exc
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/astropy/units/core.py", line 1116, in get_converter
    return self._apply_equivalencies(
           ~~~~~~~~~~~~~~~~~~~~~~~~~^
        self, other, self._normalize_equivalencies(equivalencies)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/astropy/units/core.py", line 1067, in _apply_equivalencies
    raise UnitConversionError(f"{unit_str} and {other_str} are not convertible")
astropy.units.errors.UnitConversionError: 'keV2' and 'keV' (energy/torque/work) are not convertible
```

### Screenshots

_No response_

### System Details

```
==============================
sunpy Installation Information
==============================

General
#######
OS: Ubuntu (24.04, Linux 6.8.0-49-generic)
Arch: 64bit, (x86_64)
sunpy: 6.0.3
Installation path: /tmp/tmp.u3Vb0Gt5OV/.venv/lib/python3.13/site-packages/sunpy-6.0.3.dist-info

Required Dependencies
#####################
astropy: 7.0.0
numpy: 2.1.3
packaging: 24.2
parfive: 2.1.0
pyerfa: 2.0.1.5
requests: 2.32.3

Optional Dependencies
#####################
sunpy: 6.0.3
```

### Installation method

pip

### PR DESCRIPTION
**Pull request by settwi**
Fixes #170

### CODE REVIEW

File: changelog/171.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- Update the unit attachment operator in :func:`sunkit_spex.legacy.thermal.thermal_emission`.
+  Fixes bug which caused energy units to get squared.
```
Comment (samaloney) [inline changelog/171.bugfix.rst:2]:
```suggestion
Fixes bug which caused energy units to get squared, update the way units are assigned in `~sunkit_spex.legacy.thermal.thermal_emission`.
```

File: sunkit_spex/legacy/thermal.py [hunk 2]
```diff
@@ -154,7 +154,7 @@ def setup_line_parameters(filename=None):
         "It (and its unit) therefore must be multiplied by emission measure and "
         "divided by 4 * pi * observer_distance**2 to get observed values."
     )
-    line_grid["line peaks keV"] = (line_info.peak_energy.data * line_info.attrs["units"]["peak_energy"]).to_value(
+    line_grid["line peaks keV"] = (line_info.peak_energy.data << line_info.attrs["units"]["peak_energy"]).to_value(
```
Comment (samaloney) [inline sunkit_spex/legacy/thermal.py:157]: So my guess is that probably don't even need to add the unit as the only was to get   keV$^2$ is if `line_info.peak_energy.data` is already in `keV`

**Review by samaloney — CHANGES_REQUESTED** [review]

File: sunkit_spex/legacy/thermal.py (hunk 2 above)
Comment (settwi) [reply sunkit_spex/legacy/thermal.py:157]: Right, but the prior version of astropy (?) didn't put units by default. Kind of strange

File: changelog/171.bugfix.rst (hunk 1 above)
Comment (settwi) [reply changelog/171.bugfix.rst]: Updated in latest commit

**Review by samaloney — APPROVED** [review]

**Pull request merged**